neuro_san/service/agent_main_loop.py

```diff
@@ -9,176 +9,16 @@
-        self.manifest_files = manifest_restorer.get_manifest_files()
-
-        self.agent_networks = manifest_agent_networks
-
-    def _get_default_openapi_spec_path(self) -> str:
-        """
-        Return a file path to default location of OpenAPI specification file
-        for neuro-san service.
-        """
-        file_of_class = FileOfClass(__file__, path_to_basis="../api/grpc")
-        return file_of_class.get_file_in_basis("agent_service.json")
-
-    def main_loop(self):
-        """
-        Command line entry point
-        """
-        self.parse_args()
-
-        self.server = AgentServer(self.port,
-                                  server_loop_callbacks=self,
-                                  agent_networks=self.agent_networks,
-                                  server_name=self.server_name,
-                                  server_name_for_logs=self.server_name_for_logs,
-                                  max_concurrent_requests=self.max_concurrent_requests,
-                                  request_limit=self.request_limit,
-                                  forwarded_request_metadata=self.forwarded_request_metadata)
-
-        if self.manifest_update_period_seconds > 0:
-            manifest_file: str = self.manifest_files[0]
-            updater: ManifestPeriodicUpdater =\
-                ManifestPeriodicUpdater(manifest_file, self.manifest_update_period_seconds)
-            updater.start()
-
-        # Start HTTP server side-car:
-        http_sidecar = HttpSidecar(
-            self.server.get_starting_event(),
-            self.port,
-            self.http_port,
-            self.service_openapi_spec_file,
-            self.request_limit,
-            forwarded_request_metadata=self.forwarded_request_metadata)
-        http_server_thread = threading.Thread(target=http_sidecar, args=(self.server,), daemon=True)
-        http_server_thread.start()
-
-        self.server.serve()
-
-    def loop_callback(self) -> bool:
-        """
-        Periodically called by the main server loop of ServerLifetime.
-        """
-        # Report back on service activity so the ServerLifetime that calls
-        # this method can properly yield/sleep depending on how many requests
-        # are in motion.
-        agent_services: List[GrpcAgentService] = self.server.get_services()
-        for agent_service in agent_services:
-            if agent_service.get_request_count() > 0:
-                return True
-
-        return False
+from neuro_san.service.main_loop.server_main_loop import ServerMainLoop
 
 
+# Backwards compatibility entry point
 if __name__ == '__main__':
-    AgentMainLoop().main_loop()
+    print("""
+WARNING: The class:
+    neuro_san.service.agent_main_loop.AgentMainLoop
+... has moved to be ...
+    neuro_san.service.main_loop.server_main_loop.ServerMainLoop
+Please update your Neuro SAN agent server start-up scripts accordingly.
+    """)
+    ServerMainLoop().main_loop()
```

Comment:
Another backwards compatibility.
The guys of this class has moved to neuro_san.server.main_loop.server_main_loop
This entry point and nag messages is here for backwards compatibility.